dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -162,3 +162,265 @@ def submit(self, compiled_code: str) -> None:
+            raise ValueError(
+                "workflow_job_config is required for the `workflow_job_config` submission method."
+            )
+
+    def submit(self, compiled_code: str) -> None:
+        workflow_spec = self.parsed_model["config"]["workflow_job_config"]
+        cluster_spec = self.parsed_model["config"].get("job_cluster_config", None)
+
+        # This dict gets modified throughout. Settings added through dbt are popped off
+        # before the spec is sent to the Databricks API
+        workflow_spec = self._build_job_spec(workflow_spec, cluster_spec)
+
+        self._submit_through_workflow(compiled_code, workflow_spec)
+
+    def _build_job_spec(self, workflow_spec, cluster_spec):
+        workflow_spec["name"] = workflow_spec.get('name', self.default_job_name)
+
+        cluster_settings = {}  # Undefined cluster settings defaults to serverless in the Databricks API
+        if cluster_spec is not None:
+            cluster_settings["new_cluster"] = cluster_spec
+        elif 'existing_cluster_id' in workflow_spec:
+            cluster_settings['existing_cluster_id'] = workflow_spec['existing_cluster_id']
+
+        notebook_task = {
+            'task_key': 'task_a',
+            'notebook_task': {
+                "notebook_path": self.notebook_path,
+                "source": "WORKSPACE",
+            },
+        }
+        notebook_task.update(cluster_settings)
+        notebook_task.update(workflow_spec.pop("additional_task_settings", {}))
+
+        post_hook_tasks = workflow_spec.pop("post_hook_tasks", [])
+        for task in post_hook_tasks:
+            if not 'existing_cluster_id' in task and not 'new_cluster' in task:
+                task.update(cluster_settings)
+
+        workflow_spec["tasks"] = [notebook_task] + post_hook_tasks
+        return workflow_spec
+
+    def _submit_through_workflow(self, compiled_code: str, workflow_spec) -> None:
+        self.api_client.workspace.upload_notebook(self.notebook_path, compiled_code)
+
+        job_id, is_new = self._get_or_create_job(workflow_spec)
+
+        if not is_new:
+            self._update_job(job_id, workflow_spec)
+
+        grants = workflow_spec.pop("grants", {})
+        self._update_job_permissions(job_id, grants)
+
+        run_id = self._get_or_trigger_job_run(job_id)
+
+        self.tracker.insert_run_id(run_id)
+        self.polling(
+            status_func=self.session.get,
+            status_func_kwargs={
+                "url": f"https://{self.credentials.host}/api/2.1/jobs/runs/get?run_id={run_id}",
+                "headers": self.extra_headers,
+            },
+            get_state_func=lambda response: response.json()["state"]["life_cycle_state"],
+            terminal_states=("TERMINATED", "SKIPPED", "INTERNAL_ERROR"),
+            expected_end_state="TERMINATED",
+            get_state_msg_func=lambda response: response.json()["state"]["state_message"],
+        )
+
+        run_output = self.api_client.session.get(
```

Comment:
all of these API calls are broken for now. I also need to integrate with the new polling tracker. Will get to that on Tuesday.

Reply:
Should be fixed with latest commit